### PR TITLE
Ports Whetstones

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Port of: https://github.com/tgstation/-tg-station/pull/15519

Adds in whetstones/sharpening stones.

Allows you to sharpen items that are already sharp (or rather have an edge). This adds 4 damage onto them. It can only be done once, to that item, and uses up the stone in the process. Cannot be used on items that have a force or throwforce greater than 30. Works on two-handed as well as single handed items. Two of theses starts out in the kitchen vendor

There's also an adminbus version that can be used on *anything*--it sets its damage to 200 and gives it sharp and edge.

Paradise unique features:
- Renamed to "whetstone" over "sharpening block"
 - Yes, I'm that picky =p
- Sound plays when sharpening the an item
- Can fit in your pocket
- species with claws and sharpen their claws on the stone
 - This does nothing (for now), but make a sound and give a message. If species datums are ever made non-global, then this can be added.

Semi-related
- Gave the chef 5 salt+pepper shakers so he doesn't have to worry about running out.

:cl: Fox McCloud
rscadd: Adds in whetstones for sharpening objects
tweak: Kitchen vendor starts off with 5 salt+pepper shakers
/:cl:

### CODE REVIEW

**Comment by KasparoVy** [comment]
Sharper than Einstein on adderall, hah! Wicked. Going to be interesting to see how coveted these become.

File: code/game/objects/items/weapons/whetstone.dm [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		I.sharp = 1
+	I.force = Clamp(I.force + increment, 0, max)
+	I.throwforce = Clamp(I.throwforce + increment, 0, max)
+	I.name = "[prefix] [I.name]"
+	playsound(get_turf(src), 'sound/items/Screwdriver.ogg', 50, 1)
+	name = "worn out [name]"
+	desc = "[desc] At least, it used to."
+	used = 1
+
+/obj/item/weapon/whetstone/attack_self(mob/user as mob) //This is just fluff for now. Species datums are global and not newly created instances, so we can't adjust unarmed damage on a per mob basis.
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/whetstone.dm]: and they never will be, make a hybrid unarmed system that defaults to species values and changes later based on the user if you want something like this

Comment (FlattestGuitar) [inline code/game/objects/items/weapons/whetstone.dm]: Declawing Vulpkanin finally a reality?

**Comment by Keilden** [comment]
Does this work the with replica katana?

**Comment by FlattestGuitar** [comment]
Pretty sure it only works on objects that are sharp so a replica katana isn't edgy enough.

**Comment by Fox-McCloud** [comment]
>Does this work the with replica katana?

It's a plastic toy sword---no.

**Comment by TullyBurnalot** [comment]
More salt and pepper! Sharpening sords!

:+1:

**Comment by ScratKnapp** [comment]
Teeny question. Does sharpening your claws consume the sharpening block too, or can you keep it for infinite violence-threatening-claw-sharpening action?

**Comment by Fox-McCloud** [comment]
>Teeny question. Does sharpening your claws consume the sharpening block too, or can you keep it for infinite violence-threatening-claw-sharpening action? 

Unlimited. If we can ever make it so that sharpening of claws actually does something, then the block will definitely get used up--but for now? It's infinite.

**Comment by TheDZD** [comment]
You can sharpen energy daggers, I just want to point that out.

**Comment by Fox-McCloud** [comment]
>You can sharpen energy daggers, I just want to point that out.

Can't; they have a throwforce of 35.

**Comment by TheDZD** [comment]
Ah, didn't see that it checked for throwforce.

**Pull request merged**